### PR TITLE
iwyu: Add zmq source files

### PR DESCRIPTION
**Pull request by hebasto**

### CODE REVIEW

**Review by maflcko — APPROVED** [review]
lgtm

File: src/zmq/zmqpublishnotifier.cpp [hunk 1]
```diff
@@ -6,21 +6,37 @@
...
 
 #include <zmq.h>
 
+#include <cassert>
 #include <cstdarg>
 #include <cstddef>
+#include <cstdint>
 #include <map>
 #include <optional>
+#include <string.h>
```
Comment (fanquake) [inline src/zmq/zmqpublishnotifier.cpp:32]: Given you're introducing `c*` headers everywhere else, you can use `cstring` here.

Comment (hebasto) [reply src/zmq/zmqpublishnotifier.cpp:32]: Everywhere in our codebase we use `memcpy` and `strlen` which are declared in the global namespace rather `std::` one. Therefore, the C-header `string.h` is used.

Replacements s/`memcpy`/`std::memcpy`/, s/`strlen`/`std::strlen`/ and s/`string.h`/`ctring`/ deserve its own PR, aren't they?

Comment (fanquake) [reply src/zmq/zmqpublishnotifier.cpp:32]: > Everywhere in our codebase we use memcpy and strlen which are declared in the global namespace rather std:: one.

We use both `memcpy` and `std::memcpy` in the codebase. Given we can use `memcpy` & `std::memcpy` with `cstring`, but only `memcpy` with `string.h`, and given that we are migrating towards the cpp headers, I don't see why we'd prefer `string.h`, or newly add it now, just to change it later, when we can just use `cstring` and acheive the same thing.

Comment (hebasto) [reply src/zmq/zmqpublishnotifier.cpp]: Thanks. [Updated](https://github.com/bitcoin/bitcoin/pull/26254#issuecomment-1273430487).

**Review by fanquake — APPROVED** [review]
ACK 13afcc0cd4c2975852924d2d9be5e96096147716

**Pull request merged**